### ISSUE TITLE
Incorrect quadratic expression?

### ISSUE THREAD

**Issue by MrVPlusOne** [vagmcs/Optimus#37]
Hi,

When building quadratic objectives in Optimus using the `Expression` data type, I noticed that sometimes the resulting expression seems to be incorrect. For example, the following code should result in the objective `3 * (x-2*y+3)^2`, which expands to  `27 + 18 x + 3 x^2 - 36 y - 12 x y + 12 y^2`, but optimus gives the result `obj = 12.0*x@1*x@1 + -12.0*x@0*x@1 + -54.0*x@1 + 3.0*x@0*x@0 + 27.0*x@0 + 27.0`. 

```scala
implicit val model = MPModel(SolverLib.oJSolver)

val x = MPFloatVar("x")
val y = MPFloatVar("y")
val e = x - 2.0 * y + 3.0
val obj = 3.0 * e * e
println(s"obj = $obj")
```

Any idea what caused this?

Thanks,
Jiayi

**Comment by vagmcs** [comment]
Hello @MrVPlusOne,

Thank you for the feedback. It seems to be a bug on the [product function that calculates simplifications over expression products](https://github.com/vagmcs/Optimus/blob/249210ce8166de15a3718fa330180736045dbb3d/core/src/main/scala/optimus/algebra/Expression.scala#L346) like the e * e = (x - 2 * y + 3) * (x - 2 * y + 3).

**UPDATE**: The problem is the constant 3 you add to the 'e' expression. If you change the expression e = x - 2 * y, it works fine. I think some intermediate product terms are added more than one time.

I think I fixed the issue in commit [41f88ab](https://github.com/vagmcs/Optimus/commit/41f88ab9b2b8ff31766b40f8e7b445338be0dc62). I will release a new version of Optimus today.

**Comment by vagmcs** [comment]
The issue was fixed in Optimus version 3.2.4. Please check and if everything works fine close the issue.

**Comment by MrVPlusOne** [comment]
Thanks for the update! The problem seems to be gone now.
